### PR TITLE
Delegation modal

### PR DESCRIPTION
**Pull request by bcjgit**
# TLDR

This PR adds a Modal to to make it easy to delegate votes. Button appears for users with Nouns on the `/vote` page. 

# Testing done
- Testing on desktop and mobile (both en and ja) and it looked fine
- Tested delegation on Rinkeby and it seemed to work

### CODE REVIEW

**Comment by solimander** [comment]
Starting review in the morning 🌞

File: packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      ChangeDelegateState.CHANGE_FAILURE,
+    ],
+    [
+      <NavBarButton
+        buttonText={
+          <div className={classes.delegateKVotesBtn}>
+            {locale === 'en-US ' ? (
+              <>
+                Delegate <span className={classes.highlightCircle}>{availableVotes}</span>
+                {availableVotes === 1 ? <>Votes</> : <>Votes</>}
```
Comment (solimander) [inline packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx:115]: Should be `{availableVotes === 1 ? <>Vote</> : <>Votes</>}`

File: packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+      </Trans>,
+      <Trans>
+        Your <span style={{ fontWeight: 'bold' }}>{availableVotes}</span> votes are being delegated
+        to a new account.
+      </Trans>,
+      <Trans>
+        Your <span style={{ fontWeight: 'bold' }}>{availableVotes}</span> votes have been delegated
+        to a new account.
+      </Trans>,
+      ,
```
Comment (solimander) [inline packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx:180]: nit: remove extra comma

File: packages/nouns-webapp/src/components/Proposals/index.tsx [hunk 3]
```diff
@@ -109,29 +129,19 @@ const Proposals = ({ proposals }: { proposals: Proposal[] }) => {
...
-                      <div className={classes.countdownPillContentWrapper}>
-                        <span className={classes.countdownPillClock}>
-                          <ClockIcon height={16} width={16} />
-                        </span>{' '}
-                        <span className={classes.countdownPillText}>
-                          {getCountdownCopy(p, currentBlock || 0, activeLocale)}
-                        </span>
-                      </div>
-                    </div>
+            const coundownPill = (
```
Comment (solimander) [inline packages/nouns-webapp/src/components/Proposals/index.tsx:132]: nit: `coundownPill` -> `countdownPill`

File: packages/nouns-webapp/src/wrappers/nounToken.ts [hunk 4]
```diff
@@ -146,3 +154,22 @@ export const useUserVotesAsOfBlock = (block: number | undefined): number | undef
     }) || [];
   return votes?.toNumber();
 };
+
+export const useDelegateVotes = (delagee: string) => {
```
Comment (solimander) [inline packages/nouns-webapp/src/wrappers/nounToken.ts:158]: Remove the unused param

File: packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    ChangeDelegateState.ENTER_DELEGATE_ADDRESS,
+  );
+
+  const { library, account } = useEthers();
+
+  const [delegateAddress, setDelegateAddress] = useState('');
+  const [delegateInputText, setDelegateInputText] = useState('');
+  const [delegateInputClass, setDelegateInputClass] = useState<string>('');
+  const availableVotes = useNounTokenBalance(account ?? '') ?? 0;
+  const { send: delegateVotes, state: delageeState } = useDelegateVotes(delegateAddress);
```
Comment (solimander) [inline packages/nouns-webapp/src/components/ChangeDelegatePannel/index.tsx:57]: `delageeState` -> `delegateState` or `delegateeState`

**Comment by solimander** [comment]
Hitting the point where we should dig into the network tab to see why so many external calls are being made. Load the proposals page, open the delegation modal, click 'Update Delegate' and we're already at 220 calls. Many of them are RPC calls.

**Comment by bcjgit** [comment]
> 

I think this might be independent of the modal. If I comment out the modal and load the page for an account with Nouns in it (rinkeby) I see ~300 requests after about a minute of waiting. When I reload the page with the modal I see ~ the same number (actually a bit less) and hitting the update button is only netting an incremental O(10s) of requests. 

Could be wrong but my hunch is this might not be (at least directly) related to the modal. 

--
FWIW: Even for an account without Nouns on nouns.wtf/vote I see ~150 requests many of which are RPC calls ... would be curious how this changes if the connected account holds Nouns as I found on rinkeby that when I didn't hold Nouns the number of calls was also about 100-150.

**Comment by solimander** [comment]
> I think this might be independent of the modal. If I comment out the modal and load the page for an account with Nouns in it (rinkeby) I see ~300 requests after about a minute of waiting. When I reload the page with the modal I see ~ the same number (actually a bit less) and hitting the update button is only netting an incremental O(10s) of requests.

> Could be wrong but my hunch is this might not be (at least directly) related to the modal.

Oof, okay. I was seeing an immediate 20 request increase when opening the modal, but it sounds like it's a symptom of another issue. Let's create a separate issue to diagnose and address the large number of outgoing requests.

**Review by solimander — APPROVED** [review]
Nice work! Not high priority, but may be worth adding a tooltip to notify the delegator that this will only change the voting power on proposals that haven't been created yet (i.e. no effect on pending proposals).

**Pull request merged**